docs/guide/counter/plugins/counter/counter.go

```diff
@@ -1,98 +1,197 @@
 package counter
 
 import (
-	"fmt"
+	rawerr "errors"
 
 	abci "github.com/tendermint/abci/types"
-	"github.com/tendermint/basecoin/types"
 	"github.com/tendermint/go-wire"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/modules/coin"
+	"github.com/tendermint/basecoin/stack"
+	"github.com/tendermint/basecoin/types"
 )
 
-type CounterPluginState struct {
-	Counter   int
-	TotalFees types.Coins
+// CounterTx
+//--------------------------------------------------------------------------------
+
+// register the tx type with it's validation logic
+// make sure to use the name of the handler as the prefix in the tx type,
+// so it gets routed properly
+const (
+	NameCounter = "cntr"
+	ByteTx      = 0x21
+	TypeTx      = NameCounter + "/count"
+)
+
+func init() {
+	basecoin.TxMapper.RegisterImplementation(CounterTx{}, TypeTx, ByteTx)
 }
 
 type CounterTx struct {
-	Valid bool
-	Fee   types.Coins
+	Valid    bool        `json:"valid"`
+	Fee      types.Coins `json:"fee"`
+	Sequence int         `json:"sequence"`
+}
+
+func NewCounterTx(valid bool, fee types.Coins, sequence int) basecoin.Tx {
+	return CounterTx{
+		Valid:    valid,
+		Fee:      fee,
+		Sequence: sequence,
+	}.Wrap()
```

Comment:
Yes I understand now - we need better interface referencing at the struct definition, has gotten very confusing for me, and I'm sure for others in the past/present/future, opened an issue: https://github.com/tendermint/basecoin/issues/158